### ISSUE TITLE
I already have a PGP key I have used for 2 years. I want to add it to keybase 

### ISSUE THREAD

**Issue by firecentaur** [keybase/keybase-issues#2870]
I already have a PGP key I have used for 2 years. I want to add it to keybase , how can I do this?

**Comment by songgao** [comment]
Have you looked at `keybase pgp help`?

**Comment by firecentaur** [comment]
yes but it doesn't ice an example of how to do the command

**Comment by songgao** [comment]
I think `keybase pgp import` is what you want. You can also provide a file with `keybase pgp import --infile`. See `keybase pgp import -h` for what it does exactly

**Comment by firecentaur** [comment]
thankyou, I did that, and at first it gave me an EOF error.  I fixed that
by deleting the line at the end of my infile.
Next, the keybase program prompted me for my keys password.  I entered it.
Then keybase asked me for my keybase password.  I entered that but then I
got an error on the command line:ERROR openpgp: invalid argument: no valid
signing keys




[image: circleImagePaulPreibisch.png]

*  Paul Preibisch*
  CTO: Movie English Project
  Web: https://paulpreibisch.github.io
  Email: fire@b3dmultitech.com
  LinkedIn <http://linkedIn.com/pub/paul-preibisch/4/130/a64/>

  Public PGP Key: Click Here
<https://pgp.mit.edu/pks/lookup?op=get&search=0x51DE8A676AB03950>
  Need a VPN I recommend PIA?  Sign up here: PIA
<https://www.privateinternetaccess.com/pages/buy-vpn/multitech>



*Information in this email is confidential and may be privileged. It is
intended for the addressee only. If you have received it in error, please
notify the sender immediately and delete it from your system. You should
not otherwise copy it, re-transmit it or use or disclose its contents to
anyone. *


*Although this e-mail and any attachments have been scanned for viruses,
the success of scanning products is not guaranteed. The recipient(s) should
therefore carry out any checks that they believe to be appropriate in this
respect. *
*Thank you for your co-operation.*

**Comment by songgao** [comment]
Does the file you provided include your signing key?

Also is you pgp key in the gpg keyring on the device? (If you can use gpg command without providing the key file through command line, you probably have that in the keyring already) If so `keybase pgp import` should be work

**Comment by firecentaur** [comment]
hi Song, thanks for your help...
I have a few questions.

1) Upon looking at my key file (that I generated on my computer), it looks
like:

-----BEGIN PGP PRIVATE KEY BLOCK-----
Version: GnuPG v1
random text
-----END PGP PRIVATE KEY BLOCK-----

How do I know if it includes the "Signing key"?
Also, how can I include the Signing key?

2) I am also not sure what you mean by " is you pgp key in the gpg keyring
on the device?"
I am using ubuntu, and my key is on my ubuntu key ring... is that what you
mean?



[image: circleImagePaulPreibisch.png]

*  Paul Preibisch*
  CTO: Movie English Project
  Web: https://paulpreibisch.github.io
  Email: fire@b3dmultitech.com
  LinkedIn <http://linkedIn.com/pub/paul-preibisch/4/130/a64/>

  Public PGP Key: Click Here
<https://pgp.mit.edu/pks/lookup?op=get&search=0x51DE8A676AB03950>
  Need a VPN I recommend PIA?  Sign up here: PIA
<https://www.privateinternetaccess.com/pages/buy-vpn/multitech>



*Information in this email is confidential and may be privileged. It is
intended for the addressee only. If you have received it in error, please
notify the sender immediately and delete it from your system. You should
not otherwise copy it, re-transmit it or use or disclose its contents to
anyone. *


*Although this e-mail and any attachments have been scanned for viruses,
the success of scanning products is not guaranteed. The recipient(s) should
therefore carry out any checks that they believe to be appropriate in this
respect. *
*Thank you for your co-operation.*

**Comment by songgao** [comment]
> I am using ubuntu, and my key is on my ubuntu key ring... is that what you mean?

Yes that's what I meant. Could you try `keybase pgp import` with no other argument?

**Comment by firecentaur** [comment]
Hi there Song, I tried with no arguments, and then it seems that it just
hangs there...



[image: circleImagePaulPreibisch.png]

*  Paul Preibisch*
  CTO: Movie English Project
  Web: https://paulpreibisch.github.io
  Email: fire@b3dmultitech.com
  LinkedIn <http://linkedIn.com/pub/paul-preibisch/4/130/a64/>

  Public PGP Key: Click Here
<https://pgp.mit.edu/pks/lookup?op=get&search=0x51DE8A676AB03950>
  Need a VPN I recommend PIA?  Sign up here: PIA
<https://www.privateinternetaccess.com/pages/buy-vpn/multitech>



*Information in this email is confidential and may be privileged. It is
intended for the addressee only. If you have received it in error, please
notify the sender immediately and delete it from your system. You should
not otherwise copy it, re-transmit it or use or disclose its contents to
anyone. *


*Although this e-mail and any attachments have been scanned for viruses,
the success of scanning products is not guaranteed. The recipient(s) should
therefore carry out any checks that they believe to be appropriate in this
respect. *
*Thank you for your co-operation.*
